synapse/lib/schemas.py

```diff
@@ -1164,7 +1164,7 @@
         'client_secret': {'type': 'string'},
         'client_assertion': _client_assertion_schema,
         'scope': {'type': 'string'},
-        'ssl_verify': {'type': 'boolean', 'default': True},
+        'ssl': {'type': 'object', 'properties': {'verify': {'type': 'boolean', 'default': True}}},
```

Comment:
do we need to declare the client cert/key and ca cert here and deny undeclared keys?